packages/core/src/routes/session/profile.ts

```diff
@@ -0,0 +1,56 @@
+import { usernameRegEx } from '@logto/core-kit';
+import { userInfoSelectFields } from '@logto/schemas';
+import pick from 'lodash.pick';
+import type { Provider } from 'oidc-provider';
+import { literal, object, string } from 'zod';
+
+import RequestError from '@/errors/RequestError';
+import { checkSessionHealth } from '@/lib/session';
+import koaGuard from '@/middleware/koa-guard';
+import { findUserById, updateUserById } from '@/queries/user';
+import assertThat from '@/utils/assert-that';
+
+import type { AnonymousRouter } from '../types';
+import { verificationTimeout } from './consts';
+import { checkExistingSignUpIdentifiers, checkRequiredSignUpIdentifiers } from './utils';
+
+export const profileRoute = '/session/profile';
+
+export default function profileRoutes<T extends AnonymousRouter>(router: T, provider: Provider) {
+  router.get(profileRoute, async (ctx, next) => {
+    const { accountId } = await provider.Session.get(ctx);
+
+    if (!accountId) {
+      throw new RequestError('auth.unauthorized');
+    }
+
+    const user = await findUserById(accountId);
+
+    ctx.body = pick(user, ...userInfoSelectFields);
+
+    return next();
+  });
+
+  router.patch(
+    `${profileRoute}/username`,
+    koaGuard({
+      body: object({ username: string().regex(usernameRegEx).or(literal('')).nullable() }),
+    }),
+    async (ctx, next) => {
+      const userId = await checkSessionHealth(ctx, provider, verificationTimeout);
+
+      assertThat(userId, new RequestError('auth.unauthorized'));
+
+      const { username } = ctx.guard.body;
+
+      await checkRequiredSignUpIdentifiers({ username });
```

Comment:
per the discussion we had before, seems this check on `SignInExp` is unnecessary.

Reply:
No, this is still needed. As we have to re-authenticate the user if the last authentication time is 10 minutes ago

Reply:
But the check on authentication time is done in `checkSessionHealth()`?

Reply:
Ah, sorry. My bad. You're right, the authentication time is checked in `checkSessionHealth()`.
And this function is not meant for authentication check, what it does is checking if the required identifier is missing. For example, when user calls this patch API to clear username, the server should be able to throw an error if username is a required sign-up identifier.

However, according to a recent discussion last night, we decided to prohibit the users from removing the username field. So I will remove this check from patch username API and change the body guard accordingly

Reply:
Code updated and also with some function renamings. PTAL